sql/core/src/main/java/org/apache/spark/sql/sources/v2/streaming/writer/StreamWriter.java

```diff
@@ -37,8 +41,28 @@
    */
   void commit(long epochId, WriterCommitMessage[] messages);
```

Comment:
shall we also document the requirement about idempotent？

Reply:
Good idea. PTAL at the wording